nemo/collections/asr/data/audio_to_label.py

```diff
@@ -134,24 +135,79 @@ def fixed_seq_collate_fn(self, batch):
                     sub = sig[-rem:] if rem > 0 else torch.tensor([])
                     rep_sig = torch.cat(repeat * [sig])
                     signal = torch.cat((rep_sig, sub))
+                    new_audio_lengths.append(torch.tensor(fixed_length))
                 else:
                     start_idx = torch.randint(0, chunck_len, (1,)) if chunck_len else torch.tensor(0)
                     end_idx = start_idx + fixed_length
                     signal = sig[start_idx:end_idx]
+                    new_audio_lengths.append(torch.tensor(fixed_length))
 
                 audio_signal.append(signal)
             tokens.append(tokens_i)
 
         if has_audio:
             audio_signal = torch.stack(audio_signal)
-            audio_lengths = torch.stack(audio_lengths)
+            audio_lengths = torch.stack(new_audio_lengths)
         else:
             audio_signal, audio_lengths = None, None
         tokens = torch.stack(tokens)
         tokens_lengths = torch.stack(tokens_lengths)
 
         return audio_signal, audio_lengths, tokens, tokens_lengths
 
+    def sliced_seq_collate_fn(self, batch):
+        """collate batch of audio sig, audio len, tokens, tokens len
+        Args:
+            batch (Optional[FloatTensor], Optional[LongTensor], LongTensor,
+                LongTensor):  A tuple of tuples of signal, signal lengths,
+                encoded tokens, and encoded tokens length.  This collate func
+                assumes the signals are 1d torch tensors (i.e. mono audio).
+            fixed_length (Optional[int]): length of input signal to be considered
+        """
+        slice_length = self.featurizer.sample_rate * self.time_length
+        _, audio_lengths, _, tokens_lengths = zip(*batch)
+        slice_length = min(slice_length, max(audio_lengths))
+        shift = 1 * 16000
```

Comment:
Hardcoded sample_rate? Replace with featurizer.sample_rate

Reply:
Thanks missed it, Done